test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,204 @@
+      it('gives precedence to environment variables', function () {
+        const loggerOptions = MongoLogger.resolveOptions(
+          { MONGODB_LOG_PATH: 'env' },
+          { mongodbLogPath: 'client' }
+        );
+        expect(loggerOptions.logDestination).to.equal('env');
+      });
+    });
+
+    context('maxDocumentLength', function () {
+      context('when MONGODB_LOG_MAX_DOCUMENT_LENGTH is undefined', function () {
+        it('defaults to 1000', function () {
+          const loggerOptions = MongoLogger.resolveOptions(
+            {
+              MONGODB_LOG_MAX_DOCUMENT_LENGTH: undefined
+            },
+            {}
+          );
+          expect(loggerOptions.maxDocumentLength).to.equal(1000);
+        });
+      });
+
+      context('when MONGODB_LOG_MAX_DOCUMENT_LENGTH is an empty string', function () {
+        it('defaults to 1000', function () {
+          const loggerOptions = MongoLogger.resolveOptions(
+            {
+              MONGODB_LOG_MAX_DOCUMENT_LENGTH: ''
+            },
+            {}
+          );
+          expect(loggerOptions.maxDocumentLength).to.equal(1000);
+        });
+      });
+
+      context('when MONGODB_LOG_MAX_DOCUMENT_LENGTH cannot be parsed as a uint', function () {
+        it('defaults to 1000', function () {
+          const loggerOptions = MongoLogger.resolveOptions(
+            {
+              MONGODB_LOG_MAX_DOCUMENT_LENGTH: 'invalid'
+            },
+            {}
+          );
+          expect(loggerOptions.maxDocumentLength).to.equal(1000);
+        });
+      });
+
+      context('when MONGODB_LOG_MAX_DOCUMENT_LENGTH can be parsed a uint', function () {
+        it('uses the passed value', function () {
+          const loggerOptions = MongoLogger.resolveOptions(
+            {
+              MONGODB_LOG_MAX_DOCUMENT_LENGTH: '500'
+            },
+            {}
+          );
+          expect(loggerOptions.maxDocumentLength).to.equal(500);
+        });
+      });
+    });
+
+    context('logDestination', function () {
+      context('when mongodbLogPath is undefined', function () {
+        context('when MONGODB_LOG_PATH is undefined', function () {
+          it('defaults to stderr', function () {
+            const loggerOptions = MongoLogger.resolveOptions(
+              { MONGODB_LOG_PATH: undefined },
+              { mongodbLogPath: undefined }
+            );
+            expect(loggerOptions.logDestination).to.equal('stderr');
+          });
+        });
+
+        context('when MONGODB_LOG_PATH is an empty string', function () {
+          it('defaults to stderr', function () {
+            const loggerOptions = MongoLogger.resolveOptions(
+              { MONGODB_LOG_PATH: '' },
+              { mongodbLogPath: undefined }
+            );
+            expect(loggerOptions.logDestination).to.equal('stderr');
+          });
+        });
+
+        context('when MONGODB_LOG_PATH is not an empty string', function () {
```

Comment:
```suggestion
        context('when MONGODB_LOG_PATH is a non-empty string', function () {
```
This applies to the test two below as well.